packages/flutter_tools/lib/src/globals.dart

```diff
@@ -2,6 +2,262 @@
+    }) {
+  logger.printWarning(
+    message,
+    emphasis: emphasis ?? false,
+    color: color,
+    indent: indent,
+    hangingIndent: hangingIndent,
+    wrap: wrap,
+  );
+}
+
+/// Display normal output of the command. This should be used for things like
+/// progress messages, success messages, or just normal command output.
+///
+/// Set `emphasis` to true to make the output bold if it's supported.
+///
+/// Set `newline` to false to skip the trailing linefeed.
+///
+/// If `indent` is provided, each line of the message will be prepended by the
+/// specified number of whitespaces.
+void printStatus(
+    String message, {
+      bool? emphasis,
+      bool? newline,
+      TerminalColor? color,
+      int? indent,
+      int? hangingIndent,
+      bool? wrap,
+    }) {
+  logger.printStatus(
+    message,
+    emphasis: emphasis ?? false,
+    color: color,
+    newline: newline ?? true,
+    indent: indent,
+    hangingIndent: hangingIndent,
+    wrap: wrap,
+  );
+}
+
+/// Use this for verbose tracing output. Users can turn this output on in order
+/// to help diagnose issues with the toolchain or with their setup.
+void printTrace(String message) => logger.printTrace(message);
+
+AnsiTerminal get terminal {
+  return context.get<AnsiTerminal>() ?? _defaultAnsiTerminal;
+}
+
+final AnsiTerminal _defaultAnsiTerminal = AnsiTerminal(
+  stdio: stdio,
+  platform: platform,
+  now: DateTime.now(),
+);
+
+/// The global Stdio wrapper.
+Stdio get stdio => context.get<Stdio>() ?? (_stdioInstance ??= Stdio());
+Stdio? _stdioInstance;
+
+PlistParser get plistParser => context.get<PlistParser>() ?? (
+    _plistInstance ??= PlistParser(
+      fileSystem: fs,
+      processManager: processManager,
+      logger: logger,
+    ));
+PlistParser? _plistInstance;
+
+/// The global template renderer.
+TemplateRenderer get templateRenderer => context.get<TemplateRenderer>()!;
+
+ShutdownHooks? get shutdownHooks => context.get<ShutdownHooks>();
+
+// Unless we're in a test of this class's signal handling features, we must
+// have only one instance created with the singleton LocalSignals instance
+// and the catchable signals it considers to be fatal.
+LocalFileSystem? _instance;
+LocalFileSystem get localFileSystem => _instance ??= LocalFileSystem(
+  LocalSignals.instance,
+  Signals.defaultExitSignals,
+  shutdownHooks,
+);
+
+/// Gradle utils in the current [AppContext].
+GradleUtils? get gradleUtils => context.get<GradleUtils>();
+
+CocoaPods? get cocoaPods => context.get<CocoaPods>();
+
+FlutterProjectFactory get projectFactory {
+  return context.get<FlutterProjectFactory>() ?? FlutterProjectFactory(
+    logger: logger,
+    fileSystem: fs,
+  );
+}
+
+CustomDevicesConfig get customDevicesConfig => context.get<CustomDevicesConfig>()!;
+
+PreRunValidator get preRunValidator => context.get<PreRunValidator>() ?? const NoOpPreRunValidator();
+>>>>>>> e8ec8a0dfb ([flutter_tools] Catch lack of flutter tools source missing (#93168))
```

Comment:
Merge conflict marker.

Reply:
Thanks for pointing that out. Removed it.